kurl_util/cmd/veleroplugin/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/framework"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type restoreKotsadmPlugin struct {
+	log logrus.FieldLogger
+}
+
+const name = "kurl.sh/restore-kotsadm-plugin"
+
+func main() {
+	framework.NewServer().
+		RegisterRestoreItemAction(name, newRestorePlugin).
+		Serve()
+}
+
+func newRestorePlugin(logger logrus.FieldLogger) (interface{}, error) {
+	return &restoreKotsadmPlugin{
+		log: logger,
+	}, nil
+}
+
+func (p *restoreKotsadmPlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedNamespaces: []string{"default"},
+		IncludedResources:  []string{"deployments"},
+	}, nil
+}
+
+func (p *restoreKotsadmPlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
```

Comment:
is there any way to make unit tests for functions like this?

Reply:
added